File: critter_storage/premade/Lion.java
package critter_storage.premade;
import game.*;
import java.awt.*;
import java.util.Random;

public class Lion extends Critter{

    
    public String getName(){
        return "Lion";

    }

    public Action getMove(CritterInfo info) {
        if (info.getFront() == Neighbor.OTHER) {
            return Action.INFECT;
        } else if (info.getFront() == Neighbor.WALL || info.getRight() == Neighbor.WALL){
            return Action.LEFT;
        }else if(info.getFront() == Neighbor.SAME) {
            return Action.RIGHT;
        }else{
            return Action.HOP;
        }
    }
   
    public Color getColor() {
        Random rand = new Random();
        int numOfColors = 3;
        int currentColor = rand.nextInt(numOfColors);
    

      switch(currentColor){
            default:
            return  Color.WHITE;
            
            case 0:
            return  Color.RED;

            case 1: 
            return  Color.GREEN;

            case 2: 
            return  Color.BLUE;
        }
      
    }
        
        
    
    public String toString() {
        return "L";
        
    }
}

File: CSharpVersion/CritterStorage/Premade/Lion.cs
using System;
using System.Drawing;
using CritterSimulator.Game;

namespace CritterSimulator.CritterStorage.Premade
{
    public class Lion : Critter
    {
        public override Action GetMove(ICritterInfo info)
        {
            if (info.GetFront() == Neighbor.Other)
            {
                return Action.Infect;
            }
            else if (info.GetFront() == Neighbor.Wall || info.GetRight() == Neighbor.Wall)
            {
                return Action.Left;
            }
            else if (info.GetFront() == Neighbor.Same)
            {
                return Action.Right;
            }
            else
            {
                return Action.Hop;
            }
        }

        public override Color GetColor()
        {
            Random rand = new Random();
            int currentColor = rand.Next(3);

            switch (currentColor)
            {
                case 0:
                    return Color.Red;
                case 1:
                    return Color.Green;
                case 2:
                    return Color.Blue;
                default:
                    return Color.White;
            }
        }

        public override string ToString()
        {
            return "L";
        }
    }
}
